src/vm/docs/contracts.rs

```diff
@@ -85,20 +85,70 @@ is expired, or if there's never been such a stacker, then returns none."),
         ("get-pox-info", "Returns information about PoX status.")
     ];
 
+    let bns_descriptions = vec![
+        ("namespace-preorder", "Registers the salted hash of the namespace with BNS nodes, and burns the requisite amount of cryptocurrency. Additionally, this step proves to the BNS nodes that user has honored the BNS consensus rules by including a recent consensus hash in the transaction. Returns pre-order's expiration date (in blocks)."),
+        ("namespace-reveal", "Reveals the salt and the namespace ID (pafter a namespace preorder). It reveals how long names last in this namespace before they expire or must be renewed, and it sets a price function for the namespace that determines how cheap or expensive names its will be."),
+        ("name-import", "Import name to a revealed namespace. Each imported name is given both an owner and some off-chain state."),
+        ("namespace-ready", "Launches the namespace and makes it available to the public. Once a namespace is launched, anyone can register a name in it if they pay the appropriate amount of cryptocurrency."),
+        ("name-preorder", "Preorders a name by telling all BNS nodes the salted hash of the BNS name. It pays the registration fee to the namespace owner's designated address."),
+        ("name-register", "Reveals the salt and the name to all BNS nodes, and assigns the name an initial public key hash and zone file hash."),
+        ("name-update", "Changes the name's zone file hash. You would send one of these transactions if you wanted to change the name's zone file contents. For example, you would do this if you want to deploy your own Gaia hub and want other people to read from it."),
+        ("name-transfer", "Changes the name's public key hash. You would send one of these transactions if you wanted to:
+* Change your private key
+* Send the name to someone else
+
+When transferring a name, you have the option to also clear the name's zone file hash (i.e. set it to null). This is useful for when you send the name to someone else, so the recipient's name does not resolve to your zone file."),
+        ("name-revoke", "Makes a name unresolvable. The BNS consensus rules stipulate that once a name is revoked, no one can change its public key hash or its zone file hash.  The name's zone file hash is set to null to prevent it from resolving. You should only do this if your private key is compromised, or if you want to render your name unusable for whatever reason."),
```

Comment:
Why revoking if the private key is compromised you want to transfer your name? Not clear to me.

Reply:
I copied this from the contract - not sure I have the right answer. @kantai might be able to help

Reply:
@friedger It's just one example.  The attacker could just as easily transfer your name away from you before you can revoke it.  Revoking a name is more like burning a name -- you provably destroy a name in order to make some other thing happen (e.g. imagine a future version of BNS, where you can migrate your name to the new system by burning it on the old system, which in turn grants you the right to instantiate it on the new system).